provider/python/README.md

```diff
@@ -0,0 +1,120 @@
+# Python capability provider for wasmCloud
+
+This capability provider allows wasmCloud actors to invoke functions in
+Python. It can be used to run Tensorflow, Keras, PyTorch, and other machine
+learning models in Python. Because it runs on a native host, it can take
+advantage of C libraries, including the [Intel Optimization for Tensorflow](https://www.intel.com/content/www/us/en/developer/articles/guide/optimization-for-tensorflow-installation-guide.html),
+GPUs, and any other Python libraries and extensions.
+
+This differs from other wasmCloud capability providers in a few
+significant ways:
+
+- There is not a smithy-defined interface contract. To invoke this
+  provider, an actor passes an operation name and a serialized
+  data payload. These are converted to Python objects and passed to a
+  dispatch handler function, described below.
+- Because of the limitations of the Python Global Interpreter Lock
+  (GIL), the provider is (mostly) single threaded. Some attempt is made 
+  to increase throughput by using threads and async IO for message handling and
+  serialization, but the Python functions are invoked one-at-a-time.
+- Running this provider requires a Python 3 environment to be already
+  configured on the host machine. Although the wasmCloud host could
+  download and run the provider from a remote registry, the host
+  must be configured as described below.
+
+## Invoking
+
+Here is example of a rust actor invoking the python provider to
+compute ten factorial.
+
+```rust
+use wasmbus_rpc::{common::AnySender, actor::prelude::*};
+
+async fn run(ctx: &Context) {
+    let py = AnySender::new(WasmHost::to_provider("wasmcloud:example:python", "default"));
+    let n: i32 = 10;
+    let res: i32 = py.send(&ctx, "py.factorial", &n).await?;
+    assert_eq!(res, 10 * 9 * 8 * 7 * 6 * 5 * 4 * 3 * 2);
+}
+```
+
+Here's the factorial capability provider in 5 lines of Python code:
+
+```python
+def factorial(n):
+    val = 1
+    for i in range(2, n + 1):
+        val = val * i
+    return val
+```
+
+
+## Configuration
+
+
+### Environment variables
```

Comment:
Worth pointing out here that because of these environment variable restrictions, people should be very careful about which hosts on which they run this provider. In other words, they need to know ahead of time that a given host has a valid python environment... maybe suggest adding the `python=true` label to any host that the developer knows has a valid python environment.

This environment restriction also inhibits portability and is another reason why this provider shouldn't be used in production.

Reply:
done